[Sun Yucheng] iP

Cheng20010201 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

Cheng20010201 committed:
Increment Duke to Level-1

Recustomize the chatbot name.
Add basic echo behavior indicated by the level requirement.
Recustomize the chatbot text UI.

Cheng20010201 committed:
Add a todo class to store the todo items

Cheng20010201 committed:
Todo: change from accessing from class to accessing from instances

Cheng20010201 committed:
Increment Lihua to Level 2

Using Todo class to complete the functionality in Lihua.java

Cheng20010201 committed:
Integrate a new Item class representing tasks into the application

Cheng20010201 committed:
Refactor Todo class name to make more sense

Cheng20010201 committed:
Integrate get done feature to Item and Todos

Cheng20010201 committed:
Add a class providing methods to check format of input

Cheng20010201 committed:
Add handling for index out of bound exception for
the feature of getting things done.

Cheng20010201 committed:
Fix typo in message to print out

Cheng20010201 committed:
Add get done feature to main program

Cheng20010201 committed:
Refactor with more reasonable names

Cheng20010201 committed:
Add 3 new Task's children classes

Cheng20010201 committed:
Update documentation for 3 Task classes

Cheng20010201 committed:
Update/Add helper class for user input checking and extracting

Cheng20010201 committed:
Add task adding methods in Tasks class

Cheng20010201 committed:
Add print functionality for 3 task adding

Cheng20010201 committed:
Fix bug (add error handling for format checking)

Cheng20010201 committed:
Formatting code

Cheng20010201 committed:
Use polymorphism to reduce repetitive code

Cheng20010201 committed:
Add text ui testing scripts

Cheng20010201 committed:
Add handling for wrong/missing argument of 'done'

Cheng20010201 committed:
Add exception handling, increment to level 5

Cheng20010201 committed:
Add additional testing scripts for level 5

Cheng20010201 committed:
Add a help command with Enumeration of commands

Cheng20010201 committed:
Add save functionality using JSON Simple

The task bot can only save the tasks temporarily.

A user would expect his/her tasks to be stored permanently, so that he/she can retrieve the tasks in the future.

Let's update Lihua so that the tasks can be stored in hard disk and loaded later.

Cheng20010201 committed:
Add strict date formatting for deadline

Deadline accepts all forms of string as initialization input.

Some meaningless inputs may be accepted, e.g. deadline /by haha.

Let's update the Deadline class and propagate the changes to let
deadline accepts only 'yyyy-mm-dd' format dates.

Cheng20010201 committed:
Add strict date formatting for event

Event accepts all forms of string as initialization input.

Some meaningless inputs may be accepted, e.g. event /at haha.

Let's update the Event class and propagate the changes to let
event accepts only 'yyyy-mm-dd' format dates.

Cheng20010201 committed:
Add functionality to list tasks on a specific date

The date associated with a task (if applicable) is only a record with no
utility.

User would expect a function to search tasks by date.

Let's implement this function and propogate changes.

Cheng20010201 committed:
Merge branch 'branch-Level-7' into master

Cheng20010201 committed:
Merge branch 'branch-Level-8' into master

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Tasks.java

Cheng20010201 committed:
Fix bug related to date parsing

The time field of JSON object for ToDo object is set to be an empty
string.

Calling LocalDate.parse("") would result in an exception.

Let's update the implementation so that the `LocalDate.parse()` will not
be called for ToDo objects.

Cheng20010201 committed:
Add Storage, Ui components

Ui and storage functions are embedded inside other program logic.

This will cause the code to violate OOP standards and become less readable.

Let's integrate Storage and Ui functions into classes of their own.

Cheng20010201 committed:
Add Command, Parser components

Commands are explictly handled. Understanding of command is also explicit.

This would introduce lengthy if-else block as well as repetitive String
split operations, which is not very readable, maintainable, and efficient.

Let's follow [AddressBook Level 2](https://github.com/se-edu/addressbook-level2)
 style and integrate Command and Parser classes into the application.

Cheng20010201 committed:
Remove redundant code

The application contains a lot of redundant/unused code after refactoring.

Unsued code reduces readability, and hampers the reasoning of logic.

Let's remove (part of) the redundant code from the project.

Cheng20010201 committed:
Refactor classes to form various packages

All classes are inside one directory.

Not easy to classify the classess, thus reducing readability and
hampering modulization.

By dividing classes into packages, the code becomes easier to maintain.

Let's follow a packaging style similer to [address-book level-2](https://github.com/se-edu/addressbook-level2).

Cheng20010201 committed:
Delete unused files

The four customized exception files have no implementation and are not used.

Unused files hampers readability and clarity of the whole project.

Let's delete the four files. Self-defined exception classes can be done
at a later stage.

Cheng20010201 committed:
Create simple JUnit tests on static fileds mainly

No JUnit tests are included.

No testing of important fields or methods would result in serious bugs or
unexpected behavior of code.

Let's use JUnit 5.4 to test some of the static fields of classes.

Cheng20010201 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Cheng20010201 committed:
Fix index out of bound bug for command 'done'

The index is zero-based.

Zero-based indexing may be confusing to non-programmers.

Let's modify the indexing for 'done' so that it is 1-based.

Cheng20010201 committed:
Update gradle support

The gradle class & dependency name are default.

Default names are not compatible with the project.

Let's update build.gradle so that the dependency and class name are supported.

Cheng20010201 committed:
Add javadoc for all classes

No/incomplete javadoc is provided.

Missing javadoc would affect the readability and maintenance of the code.

Let's add javadoc to our application.

Cheng20010201 committed:
Tweak the code to follow code standards

The code style is mostly correct, with few exceptions because of carelessness.

Making the code style standard is better.

Let's update the style a bit so that the code basically abides the standard.

Cheng20010201 committed:
Modify welcome and good back emonji

The emonji is not displaying properly in windows terminal.

Let's update the emonji so that it gets displayed properly.

Cheng20010201 committed:
Add find by keyword functionality

The application only supports list tasks by time (if applicable).

The user would want to search tasks by name/keyword.

Let's integrate the find by keyword functionality into our application.

Cheng20010201 committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc' into master

Cheng20010201 committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard' into master

# Conflicts:
#	src/main/java/lihua/commands/AddCommand.java
#	src/main/java/lihua/commands/DeleteCommand.java
#	src/main/java/lihua/commands/DoneCommand.java
#	src/main/java/lihua/commands/ExitCommand.java
#	src/main/java/lihua/commands/HelpCommand.java
#	src/main/java/lihua/commands/ListCommand.java
#	src/main/java/lihua/commons/Messages.java

Cheng20010201 committed:
Merge remote-tracking branch 'origin/branch-Level-9' into master

# Conflicts:
#	src/main/java/lihua/commands/ListCommand.java
#	src/main/java/lihua/commons/Messages.java
#	src/main/java/lihua/tasks/Task.java
#	src/main/java/lihua/tasks/Tasks.java

Cheng20010201 committed:
Fix issues in regression testing

After adding the find functionality, the regression testing fails

Testing failure means potential bugs.

Let's fix the regression testing issue.

Cheng20010201 committed:
Add checkstyle files and make style changes

The code style check is done manually through eyeballs.

Doing code style check manually is time-consuming and error-prone.

Let's integrate auto-checkstyle using gradle.

Cheng20010201 committed:
Fix issues with jar file using gradle

After the code is refactored, the jar file encounters "ClassNotFoundException" upon start.

This is a bug caused by java file packaging.

Let's fix this issue by unpackage Lihua.java file.

Cheng20010201 committed:
Enable assertion in runtime

Assertion in runtime is not supported by build.gradle.

Sometimes such assertion is necessary to ensure correctness of program.

Let's configure the build.gradle file to allow assertions.

Cheng20010201 committed:
Merge branch 'branch-A-Gradle'

Cheng20010201 committed:
Integrate javaFX tutorial GUI into Lihua

Lihua is implemented in a CLI application way.

A GUI will make the app easire to use for users, as users can see
the feedback in a more friendly manner.

Although it is far from complete, let's integrate the GUI features the
javaFX tutorial (https://se-education.org/guides/tutorials/javaFx.html)
introduces into the current Lihua project.

Cheng20010201 committed:
Refactor classes to fit in different packages

All newly created GUI classes are put under package lihua directly.

This arrangement of classes are not easy to maintain and manage.

It's better to integrate all classes into different packages, based on their functionalities.

Cheng20010201 committed:
Integrate Lihua's original functionality into tutorial GUI

The tutorial GUI only supports echo functionality.

Echo functionality is, of course, not adequate.

Let's integrate almost all Lihua's original functionalities into the naive GUI.
The exit command is not supported right now since the mechanism of
exit command is different from normal ones.

Cheng20010201 committed:
Increment GUI

The GUI style follows the tutorial style.

The GUI is not unique to Lihua.

Let's chaneg some default style setting of GUI so that Lihua looks unique.

Cheng20010201 committed:
Merge branch 'branch-Level-10'

Cheng20010201 committed:
Add minor changes to GUI; update gradle JavaFX version

ZhangAnli commented on src/main/java/lihua/commands/FindCommand.java:
```diff
@@ -0,0 +1,33 @@
+ * Command class representing a command to find tasks by keyword.
+ */
+public class FindCommand extends Command {
+    /** Command word for find command */
```
Do you think it would help if you left spaces between consecutive methods? Have noticed this in a few places and because of the javadoc comments, it may make your code seem a bit "cramped".

Cheng20010201 replied:
Sorry correct me if I misunderstood your point🤔. Are you suggesting like I leave spaces of more than one line between consecutive methods to make the code look more 'sparse'?

Eriksen2411 replied:
For me, I think the spacing between consecutive methods is good enough. I just think that at the beginning of the class ( between attributes and constructor) you should leave a one line spacing for easier looking

ZhangAnli commented on src/main/java/lihua/tasks/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+ * Represents a lihua.tasks.Deadline item, which is a child of lihua.tasks.Task
+ */
+public class Deadline extends Task {
+    private final LocalDate by;
```
Perhaps a more intuitive name for the localdate variable will be better? I feel that just stating "by" could be hard to read for some.

Cheng20010201 replied:
like `deadlineDate`? 🤔

Eriksen2411 replied:
Hmmm i also agree with ZhangAnli this point. and deadlineDate seems good enough.

branzuelajohn replied:
I agree with Anli and Eriksen regarding this

ZhangAnli commented on src/test/java/lihua/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,4 @@
+package lihua.tasks;
+
+public class DeadlineTest {
```
Would like to clarify the purpose of this file since it is essentially an empty file?

Cheng20010201 replied:
Oops I added the test class but forgot to add content to it. Thanks for pointing it out!

Eriksen2411 replied:
Also agree :)) Just for me to add 1 comment. TY

ZhangAnli reviewed (commented):
Overall, the code was written well and did not violate much coding violations. Good job!

Eriksen2411 commented on src/main/java/lihua/commands/HelpCommand.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class HelpCommand extends Command {
+    /** Command word for help command */
+    public static final String COMMAND_WORD = "help";
```
Do we need COMMAND_WORD and MESSAGE_USAGE here at the same time both serving as attributes. I think you should only keep COMMAND_WORD as attribute and the other could be converted to method return.

branzuelajohn commented on src/main/java/lihua/tasks/Task.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    /**
+     * Set the status of the task; e.g. done or not done
```
"Sets" instead of set?

branzuelajohn commented on src/test/java/lihua/tasks/EventTest.java:
```diff
@@ -0,0 +1,4 @@
+package lihua.tasks;
+
+public class EventTest {
```
Are you supposed to add content to this test class file, because this class does not serve a purpose now as it is empty

branzuelajohn commented on src/main/java/lihua/main/Main.java:
```diff
@@ -0,0 +1,33 @@
+import lihua.ui.MainWindow;
+
+/**
+ * A GUI for Duke using FXML.
```
should this be "A GUI for Lihua..." to be consistent with the name you decided to name your bot

Cheng20010201 committed:
Fix styling issues pointed out by peers

The styling is mostly ok, except some minor details.

Peers pointed the details out for me during PR review practice.

Let's update the code style to comply to good java code standards.

Cheng20010201 committed:
Add assertions to Lihua

Assertions do not exists anywhere in code.

Liberal use of assertions can reduce the likelyhood of bugs.

Let's integrate some important assertions we make about the code in Lihua.

Cheng20010201 committed:
Refactor code to make the code have better code quality

The code follows basic java coding standards, but some of the code
is written in a way that does not follow best practices of code quality.

Code of poor code quality can propogate the messy effects to the future.

Let's refactor the code to make it have better code quality.

Cheng20010201 committed:
Refactor code to make the code have better code quality

The code follows basic java coding standards, but some of the code
is written in a way that does not follow best practices of code quality.

Code of poor code quality can propogate the messy effects to the future.

Let's refactor the code to make it have better code quality.

Cheng20010201 committed:
Merge pull request #1 from Cheng20010201/branch-A-Assertions

Add assertions to Lihua

Cheng20010201 committed:
Merge branch 'master' into branch-A-CodeQuality

Cheng20010201 committed:
Merge pull request #2 from Cheng20010201/branch-A-CodeQuality

Add code quality improvements

Cheng20010201 committed:
Add CI to the project

CI is not integrated into the project.

If CI is integrated, push/pr operations will be simpler.

Let's add CI to Lihua.

Cheng20010201 committed:
Fix a bug in data storage

Incorrect exception handling leads to the console to print error information
when initially there is no data file.

This should not happen since no data file initially should be a normal
situation instead of an error.

Let's update Storage.java to fix this bug.

Cheng20010201 committed:
Remove all text-ui-test files

The text-ui-test files are to be used to test the text ui of the application.

This testing is no longer needed since there is already a GUI available.

Let's drop related files to make our workspace cleaner.

Cheng20010201 committed:
Add C-Sort functionality to the project

The bot lists down task in a FIFO order.

A more natural order may be order the tasks by name. Also, if additonal
tag -time is provided to command 'list', all the tasks with time information
will be listed in chronological order.

Let's integrate this feature and propograte changes.

Cheng20010201 committed:
Merge branch 'branch-BCD-Extension'

Cheng20010201 committed:
Give credit to reused code for GUI

The code for GUI, though taken from the tutorials provided, is not given credit.

It is best to give credit to the author of them.

Let's give credit to the author for code snippets taken from him.

Cheng20010201 committed:
Fix issues in regressions

After previous incrementation or  sorting functionality, the current test cases fail.

Failure in test cases indicates potential bugs, or some regression issues. It is better
to handle them as soon as possible.

Let's fix the test (mainly for message testing) for the project.

Cheng20010201 committed:
Refactor code by arranging packages in a more reasonable manner

Cheng20010201 committed:
Add streams to some simple task loop handling in Tasks.java

Cheng20010201 committed:
Change the directory of gradle.yml to follow the tutorials of CI

Cheng20010201 committed:
Update gradle.yml to trim unnecessary parts for the project

Cheng20010201 committed:
Update styles to conform with CheckStyle

Cheng20010201 committed:
Merge branch 'master' into branch-A-Streams

Cheng20010201 committed:
Merge pull request #3 from Cheng20010201/branch-A-Streams

merge branch-A-Streams with master branch

Cheng20010201 committed:
Reset the dialog box's maximum height

The dialog box maximum height is the default value.
When too much text comes into the dialog box, it is possible for the dialog
box not to contain all information.

The dialog should be able to auto-adjust the maximum height
with the text it is displaying.

Let us update the dialog box so that the desired functionality is achieved.

Cheng20010201 committed:
Add different backgrounds for Lihua and user

The background is the same for both enitites.

It will be more distinguishable if the backgrounds are different.

Let us update the background of dialog box so that Lihua and user
have different background colors.

Cheng20010201 committed:
Beautify the profile icon

The icon is directly taken from a picture.

The GUI would be more beautiful if the icon can have a rounded, colored, shaded effect.

Let us update the GUI for dialog to achieve this effect.

Cheng20010201 committed:
Refactor GUI code to remove unused parts

Cheng20010201 committed:
Modify the background color for GUI

Cheng20010201 committed:
Let bot change background color when the user input is incorrect to remind user

The bot does not change background color.

User may want to be reminded if their input is not in the correct format.

Let us change the code so that the bot can change background color if the
user input is incorrect. Right now we do not care how the checking if input
is correct, but we focus on how to change color. The checking will be done in
the next commit.

Cheng20010201 committed:
Update Lihua to allow CommandResult containing a field indicating of whether
the user input is in the correct format or not

The background judge of Lihua is always going to give an 'error' background.

We want our Lihua to give 'error' background only if the user input has an 'error'.

Let us update CommandResult and propagate changes to accomodate the functionality.

Cheng20010201 committed:
Merge branch 'branch-A-BetterGui'

Cheng20010201 committed:
Refactor code to follow check style guidance

Cheng20010201 committed:
Set theme jekyll-theme-cayman

Cheng20010201 committed:
Update README.MD of ip/docs

Cheng20010201 committed:
Update README.MD of ip/docs

Cheng20010201 committed:
Merge branch 'master' of github.com:Cheng20010201/ip

Cheng20010201 committed:
Make tasks listed in alphabetic ordering

Cheng20010201 committed:
Update ip/docs/README.md for the product website

The product website is not complete.

The product webstie will provide enough information if it is complete.

Let us update ip/docs/README.md and complete missing information.

Cheng20010201 committed:
Fix CI issues resulting from styling

Cheng20010201 committed:
Fix README issues (table not correctly rendered in published view)

Cheng20010201 committed:
Fix README issues (table not correctly rendered in published view)

Cheng20010201 committed:
Fix README issues (table not correctly rendered in published view)

Cheng20010201 committed:
Update README.md

Cheng20010201 committed:
Update release link

Cheng20010201 committed:
Fix formatting issues for README.md

Cheng20010201 committed:
Fix hyper link issues for README.md

Cheng20010201 committed:
Fix hyperlink issues for README.md

Cheng20010201 committed:
Add javadoc for previously missed public classes and methods
